source/codal_target_hal.cpp

```diff
@@ -51,8 +51,23 @@ int target_random(int max)
     return codal::random(max);
 }
 
-void target_wait_us(unsigned long us) {
-    codal::system_timer_wait_us(us);
+void target_wait_us(unsigned long us)
+{
+    if (us == 0)
+        return;
+
+    if (us < 3)
+        us = 3;
+
+    uint32_t n = (uint32_t)(us - 3) * (CODAL_CPU_CLOCK_SPEED / 1000000) / 3;
```

Comment:
`n` will be `0` for `target_wait_us(3)` and the loop will spin forever, well a few seconds